lib/ansible/modules/cloud/amazon/cloudfront_origin_access_identity.py

```diff
@@ -0,0 +1,285 @@
+description:
+    - Allows for easy creation, updating and deletion of origin access
+      identities.
+requirements:
+  - boto3 >= 1.0.0
+  - python >= 2.6
+version_added: "2.4"
+author: Willem van Ketwich (@wilvk)
+options:
+    state:
+      description:
+        - The state of the resource. Valid states are
+            present
+            absent
+      required: false
+      default: present
+    origin_access_identity_id:
+      description:
+        - The origin_access_identity_id of the cloudfront distribution.
+      required: false
+    comment:
+      description:
+        - A comment to describe the cloudfront origin access identity.
+      required: false
+    caller_reference:
+      description:
+        - A unique identifier to reference the origin access identity by.
+      required: false
+
+'''
+
+EXAMPLES = '''
+
+# create an origin access identity
+
+- cloudfront_origin_access_identity:
+    state: present
+    caller_reference: this is an example reference
+    comment: this is an example comment
+
+# update an existing origin access identity
+# using caller_reference as an identifier
+
+- cloudfront_origin_access_identity:
+     state: present
+     origin_access_identity_id: E17DRN9XUOAHZX
+     caller_reference: this is an example reference
+     comment: this is a new comment
+
+# delete an existing origin access identity
+# using caller_reference as an identifier
+
+- cloudfront_origin_access_identity:
+     state: absent
+     caller_reference: this is an example reference
+     comment: this is a new comment
+
+'''
+
+RETURN = '''
+
+location:
+    description: describes a url specifying the origin access identity.
+    returned: always
+    type: str
+
+'''
+
+from ansible.module_utils.ec2 import get_aws_connection_info, ec2_argument_spec
+from ansible.module_utils.ec2 import boto3_conn, HAS_BOTO3
+from ansible.modules.cloud.amazon.cloudfront_facts import (
+    CloudFrontFactsServiceManager)
+import ansible.module_utils.cloudfront as helpers
+from ansible.module_utils.ec2 import camel_dict_to_snake_dict
+from ansible.module_utils.basic import AnsibleModule
+from botocore.signers import CloudFrontSigner
+import datetime
+from functools import partial
+import json
+import traceback
+
+try:
+    import botocore
+except ImportError:
+    pass
+
+
+class CloudFrontOriginAccessIdentityServiceManager(object):
+    """
+    Handles cloudfront origin access identity service calls to aws
+    """
+
+    def __init__(self, module):
+        self.module = module
+        self.create_client('cloudfront')
+
+    def create_client(self, resource):
+        try:
+            region, ec2_url, aws_connect_kwargs = get_aws_connection_info(
+                self.module, boto3=True)
+            self.client = boto3_conn(
+                self.module, conn_type='client', resource=resource,
+                region=region, endpoint=ec2_url, **aws_connect_kwargs)
+        except botocore.exceptions.NoRegionError:
```

Comment:
boto3_conn() now handles NoRegionError and ClientError so you can remove that here.